src/DGtal/geometry/volumes/BoundedLatticePolytope.h

```diff
@@ -0,0 +1,878 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file BoundedLatticePolytope.h
+ * @author Jacques-Olivier Lachaud (\c jacques-olivier.lachaud@univ-savoie.fr )
+ * Laboratory of Mathematics (CNRS, UMR 5127), University of Savoie, France
+ *
+ * @date 2020/01/02
+ *
+ * Header file for module BoundedLatticePolytope.cpp
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(BoundedLatticePolytope_RECURSES)
+#error Recursive header files inclusion detected in BoundedLatticePolytope.h
+#else // defined(BoundedLatticePolytope_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define BoundedLatticePolytope_RECURSES
+
+#if !defined BoundedLatticePolytope_h
+/** Prevents repeated inclusion of headers. */
+#define BoundedLatticePolytope_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <list>
+#include <vector>
+#include <string>
+#include "DGtal/base/Common.h"
+#include "DGtal/kernel/CSpace.h"
+#include "DGtal/kernel/domains/HyperRectDomain.h"
+#include "DGtal/arithmetic/IntegerComputer.h"
+#include "DGtal/arithmetic/ClosedIntegerHalfPlane.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // template class BoundedLatticePolytope
+  /**
+     Description of template class 'BoundedLatticePolytope' <p> \brief
+     Aim: Represents an nD polytope, i.e. a convex polyhedron, as a
+     set of inequalities. Otherwise said, it is a H-representation of
+     a polytope (as an intersection of half-spaces). A limitation is
+     that we model only bounded polytopes, i.e. polytopes that can be
+     included in a finite bounding box.
```

Comment:
maybe you should add a few comments on the arithmetic: this classisms dedicated to rational polytopes on the Point::Coordinate Euclidean ring. Am I right?

Reply:
I am not sure of what you mean. H and V reps are classical even with real coordinates vertices. And BoundedLatticePolytope is limited to lattice point vertices.